Src/Newtonsoft.Json/JsonWriter.Async.cs

```diff
@@ -1680,114 +1680,122 @@ protected Task SetWriteStateAsync(JsonToken token, object value, CancellationTok
-                    return writer.WriteValueAsync(value == null ? (uint?)null : (uint)value, cancellationToken);
-                case PrimitiveTypeCode.Int64:
-                    return writer.WriteValueAsync((long)value, cancellationToken);
-                case PrimitiveTypeCode.Int64Nullable:
-                    return writer.WriteValueAsync(value == null ? (long?)null : (long)value, cancellationToken);
-                case PrimitiveTypeCode.UInt64:
-                    return writer.WriteValueAsync((ulong)value, cancellationToken);
-                case PrimitiveTypeCode.UInt64Nullable:
-                    return writer.WriteValueAsync(value == null ? (ulong?)null : (ulong)value, cancellationToken);
-                case PrimitiveTypeCode.Single:
-                    return writer.WriteValueAsync((float)value, cancellationToken);
-                case PrimitiveTypeCode.SingleNullable:
-                    return writer.WriteValueAsync(value == null ? (float?)null : (float)value, cancellationToken);
-                case PrimitiveTypeCode.Double:
-                    return writer.WriteValueAsync((double)value, cancellationToken);
-                case PrimitiveTypeCode.DoubleNullable:
-                    return writer.WriteValueAsync(value == null ? (double?)null : (double)value, cancellationToken);
-                case PrimitiveTypeCode.DateTime:
-                    return writer.WriteValueAsync((DateTime)value, cancellationToken);
-                case PrimitiveTypeCode.DateTimeNullable:
-                    return writer.WriteValueAsync(value == null ? (DateTime?)null : (DateTime)value, cancellationToken);
-                case PrimitiveTypeCode.DateTimeOffset:
-                    return writer.WriteValueAsync((DateTimeOffset)value, cancellationToken);
-                case PrimitiveTypeCode.DateTimeOffsetNullable:
-                    return writer.WriteValueAsync(value == null ? (DateTimeOffset?)null : (DateTimeOffset)value, cancellationToken);
-                case PrimitiveTypeCode.Decimal:
-                    return writer.WriteValueAsync((decimal)value, cancellationToken);
-                case PrimitiveTypeCode.DecimalNullable:
-                    return writer.WriteValueAsync(value == null ? (decimal?)null : (decimal)value, cancellationToken);
-                case PrimitiveTypeCode.Guid:
-                    return writer.WriteValueAsync((Guid)value, cancellationToken);
-                case PrimitiveTypeCode.GuidNullable:
-                    return writer.WriteValueAsync(value == null ? (Guid?)null : (Guid)value, cancellationToken);
-                case PrimitiveTypeCode.TimeSpan:
-                    return writer.WriteValueAsync((TimeSpan)value, cancellationToken);
-                case PrimitiveTypeCode.TimeSpanNullable:
-                    return writer.WriteValueAsync(value == null ? (TimeSpan?)null : (TimeSpan)value, cancellationToken);
+            for (;;)
```

Comment:
I prefer `while (true)`

Reply:
Ah yes, I recall that you do. (It'll probably never not seem strange to me)